src/treemodel/rootelement.js

```diff
@@ -19,16 +19,35 @@ export default class RootElement extends Element {
 	 *
 	 * @param {engine.treeModel.Document} doc {@link engine.treeModel.Document} that is an owner of the root.
 	 * @param {String} name Node name.
+	 * @param {String} rootName Root name inside parent {@link engine.treeModel.Document}.
 	 */
-	constructor( doc, name ) {
+	constructor( doc, name, rootName ) {
 		super( name );
 
 		/**
 		 * {@link engine.treeModel.Document} that is an owner of this root.
 		 *
 		 * @readonly
-		 * @member {engine.treeModel.Document} engine.treeModel.RootElement#doc
+		 * @member {engine.treeModel.Document} engine.treeModel.RootElement#document
 		 */
 		this.document = doc;
+
+		/**
+		 * Name of this root inside {@link engine.treeModel.Document} that is an owner of this root.
+		 *
+		 * @readonly
+		 * @member {String} engine.treeModel.RootElement#rootName
+		 */
+		this.rootName = rootName;
+	}
+
+	/**
+	 * Custom toJSON method to solve child-parent circular dependencies.
+	 *
+	 * @method engine.treeModel.RootElement#toJSON
+	 * @returns {String} Name of this root inside {@link engine.treeModel.Document} that is an owner of this root.
+	 */
+	toJSON() {
```

Comment:
Implementation of this method is confusing. Why is the name of the root enough in `toJSON()` to represent the root? Why there are no children or attributes of it (in fact – attributes are really missing)?

If some of these things are not necessary, this should be documented why. The fact that this implementation solve child-parent circ deps is a secondary aspect, and I'm not even sure whether it's true (because the same would need to be done for all elements, while in those cases it's handled by deleting `child.parent`).